dotnet log-trace-injection updates (#9856)
DataDog/documentation: content/en/tracing/connect_logs_and_traces/dotnet.md

  </layout>
```

For NLog version 4.5:

```xml
 <!-- Add includeMdc="true" to emit MDC properties -->
  <layout xsi:type="JsonLayout" includeMdc="true">
    <attribute name="date" layout="${longdate}" />
    <attribute name="level" layout="${level:upperCase=true}"/>
    <attribute name="message" layout="${message}" />
    <attribute name="exception" layout="${exception:format=ToString}" />
  </layout>
```

> For additional examples, see the automatic trace ID injection projects [for NLog45][1] or [for NLog46][2] on GitHub.

Reviewer: ```suggestion
For additional examples, see the automatic trace ID injection projects using [NLog 4.5][1] or [NLog 4.6][2] on GitHub.
```
What do you think of this change?

Author: Yup, that looks much better.

[1]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog45Example/NLog.config
[2]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog46Example/NLog.config
{{% /tab %}}
{{< /tabs >}}

## Manually inject trace and span IDs

**Note:** If you are not using a [Datadog Log Integration][7] to parse your logs, custom log parsing rules must parse `dd.trace_id` and `dd.span_id` as strings. For information, see the [FAQ on this topic][8].

To manually correlate your [APM traces][9] with application logs:

1. Reference the [`Datadog.Trace` NuGet package][10] in your project.

2. Use the `CorrelationIdentifier` API to retrieve correlation identifiers and add them to the log context while a span is active.

> 3. Configure mapped diagnostic context (MDC) to use the following injected keys:

Reviewer: The setup with config files for log4net and NLog, and the `.Enrich.FromLogContext()` for Serilog are required for both automatic AND manual instrumentation. What do you think would be the best way to reference those from both sets of instructions?

Author: Got it, update incoming.

    | Required key   | Description                                  |
    | -------------- | -------------------------------------------- |
    | `dd.env`       | Globally configures the `env` for the tracer. Defaults to `""` if not set. |

>     | `dd.service`   | Globally configures the root service name. Ddefaults to the name of the application or IIS site name if not set.  |

Reviewer 2: ```suggestion
    | `dd.service`   | Globally configures the root service name. Defaults to the name of the application or IIS site name if not set.  |
```

    | `dd.version`   | Globally configures `version` for the service. Defaults to `""` if not set.  |
    | `dd.trace_id`  | Active trace ID during the log statement. Defaults to `0` if no trace.  |

>     | `dd.span_id`   | Active span ID during the log statement. Default to 0 if no trace. |

Reviewer 2: ```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Default to `0` if no trace. |
```

>     | `dd.span_id`   | Active span ID during the log statement. Default to 0 if no trace. |

Author: ```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Defaults to `0` if no trace. |
```

>     | `dd.span_id`   | Active span ID during the log statement. Default to 0 if no trace. |

Author: ```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Defaults to 0 if no trace. |
```

>     | `dd.span_id`   | Active span ID during the log statement. Default to `0` if no trace. |

Author: ```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Defaults to `0` if no trace. |
```

  </layout>
```

For NLog version 4.5:

```xml
 <!-- Add includeMdc="true" to emit MDC properties -->
  <layout xsi:type="JsonLayout" includeMdc="true">
    <attribute name="date" layout="${longdate}" />
    <attribute name="level" layout="${level:upperCase=true}"/>
    <attribute name="message" layout="${message}" />
    <attribute name="exception" layout="${exception:format=ToString}" />
  </layout>
```

> For additional examples, see the automatic trace ID injection projects using [NLog 4.5][1] or [NLog 4.6][2] on GitHub.

Reviewer: ```suggestion
For additional examples, see the automatic trace ID injection projects using [NLog 4.0][1], [NLog 4.5][2], or [NLog 4.6][3] on GitHub.
```

I just recently added an NLog 4.0 example 😄

> [1]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog45Example/NLog.config
> [2]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog46Example/NLog.config

Reviewer: ```suggestion
[1]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog40Example/NLog.config
[2]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog45Example/NLog.config
[3]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog46Example/NLog.config
```
I just recently added an NLog 4.0 example 😄
